nav-app/src/app/techniques-search/techniques-search.component.ts

```diff
@@ -110,12 +181,50 @@ export class TechniquesSearchComponent implements OnInit {
         this.viewModel.unselectTechniqueAcrossTactics(technique);
     }
 
-    public selectAll(): void {
-        for (let result of this.results) this.select(result);
+    public selectAll(items: any[]): void {
+        for (let result of items) this.select(result);
     }
 
-    public deselectAll(): void {
-        for (let result of this.results) this.deselect(result);
+    public deselectAll(items: any[]): void {
+        for (let result of items) this.deselect(result);
     }
 
+    public selectStix(stixObject: BaseStix): void {
+        for (let technique of this.getRelated(stixObject)) {
+            this.viewModel.selectTechniqueAcrossTactics(technique);
+        }
+    }
+
+    public deselectStix(stixObject: BaseStix): void {
+        for (let technique of this.getRelated(stixObject)) {
+            this.viewModel.unselectTechniqueAcrossTactics(technique);
+        }
+    }
+
+    public selectAllStix(items: BaseStix[]): void {
+        for (let stixObject of items) {
+            this.selectStix(stixObject);
+        }
+    }
+
+    public deselectAllSitx(items: BaseStix[]): void {
+        for (let stixObject of items) {
+            this.deselectStix(stixObject);
+        }
+    }
```

Comment:
Again I think this is a misuse of the word Stix. Also STIX is misspelled in `deselectAllSitx`